c++/include/orc/Reader.hh

```diff
@@ -657,6 +657,12 @@ namespace orc {
      * @param rowNumber the next row the reader should return
      */
     virtual void seekToRow(uint64_t rowNumber) = 0;
+
+    /**
+     * Get the current stripe position entries for the specified column.
+     * @return the position entries for the specified column.
+     */
+    virtual std::vector<std::vector<int>> getCurrentStripePositionEntries(uint64_t columnId) = 0;
```

Comment:
Why this is placed in `RowReader` but not `Reader` which is more appropriate for reading metadata? In addition, it would be more useful if we can query the positions for any combination of columns from any stripe, which may help I/O optimization for index streams.